Source/SubtitleProviderSsaAss.h

```diff
@@ -469,27 +487,80 @@ class SubtitleProviderSsaAss : public SubtitleProvider
                                     break;
                                 }
 
+                                // Absolute positioning does not change the width (normally)
+                                // MarginL and MarginR still apply, which we cannot replicate
+                                // in all cases
+                                auto padding1 = subRegion.Padding();
+
                                 switch (subStyle.LineAlignment())
                                 {
                                 case  TimedTextLineAlignment::Start:
                                     pos.X = x * 100;
                                     extent.Width = (1.0 - x) * 100;
+
+                                    // Both margins have an effect, but a left margin here doesn't change
+                                    // the alignment point, so we add the left margin to the right margin
+                                    // and set left to zero.
+                                    padding1.After += padding1.Before;
+                                    padding1.Before = 0;
+
+                                    // Now subtract the reduced width from the padding (which is meant to
+                                    // be applied to a 100% width box)
+                                    padding1.After -= (100 - extent.Width);
```

Comment:
I do not really understand why the right margin needs to be reduced here (and below). But I guess you tested this with actual files, comparing with how Aegisub renders them, right?

Reply:
Yes I did. You can easily do for yourself like this:

In Aegisub, open any ASS file and then create a dummy video for it. 
Detach the video window (just for ease).
Double-Click an event.
Then you get there:

![image](https://github.com/ffmpeginteropx/FFmpegInteropX/assets/4985349/3436b46d-3f9c-4330-9931-5d5d006d6678)

By double-clicking, you can set the position.
By adding {\anX}, you can change the alignment.
And now you can play with different margins:

![image](https://github.com/ffmpeginteropx/FFmpegInteropX/assets/4985349/0cb889cf-fb74-4c38-99a8-3731e000394d)


You'll see that Aegisub always calculates with a width of 100% and then it applies the L/R margins, but when left or right aligned, it adds both margins to the opposite side.

Since we cannot have a width of 100% but need to reduce it so the region fits into the video frame, we need to subtract this reduced width from the margins (because it's like as if we had already applied this reduced width as a margin).